CMakeLists.txt

```diff
@@ -214,13 +219,6 @@ else()
     add_compile_options(-DDEBUG -g3 -gdwarf-3 --coverage)
   endif()
 
-  if (TILEDB_SERIALIZATION OR
-     (CMAKE_COMPILER_IS_GNUCC AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0))
-    # CapnProto 0.6.1 contains some unused parameter warnings that we need to
-    # ignore in order to allow for Werror builds.
-    add_compile_options(-Wno-unused-parameter)
-  endif()
```

Comment:
I see the comment references capnproto, but the actual condition was not specific just to TILEDB_SERIALIZATION, 
should this be left active with the second half of the 'OR' condition?

Reply:
I can put it back if there's a strong objection, but in the limited situations where we can still build with GCC < 5 (don't think any of the cloud SDKs work anymore), IMO it is ok to require disabling Werror.

Reply:
I'm ok with it, just asking to be sure you'd considered.